packages/experimental/Shimmer/src/Shimmer.tsx

```diff
@@ -0,0 +1,97 @@
+import { ShimmerElementType, shimmerName, ShimmerProps, ShimmerType } from './Shimmer.types';
+import { compose, mergeProps, withSlots, UseSlots, buildUseStyling } from '@fluentui-react-native/framework';
+import { Animated } from 'react-native';
+import { stylingSettings } from './Shimmer.styling';
+
+export function useShimmerAnimation(memoData: any) {
+  let startValue = useState(new Animated.Value(0))[0]
+  useEffect(() => {
+    Animated.loop(
+      Animated.sequence([
+        Animated.timing(startValue, {
+          toValue: memoData.toValue,
+          duration: memoData.duration,
+          delay: memoData.delay,
+          useNativeDriver: false,
+        })]),
+    ).start()
+  })
+  return startValue;
+}
+
+const useStyling = buildUseStyling(stylingSettings);
+export const Shimmer = compose<ShimmerType>({
+  displayName: shimmerName,
+  ...stylingSettings,
+  slots: {
+    root: Svg,
+    rect: Rect,
+    circle: Circle,
+    image: Image,
+  },
+  render: (props: ShimmerProps, useSlots: UseSlots<ShimmerType>) => {
+    const Slots = useSlots(props);
+    let AnimatedLinearGradient = Animated.createAnimatedComponent(LinearGradient);
+    const tokens = useStyling(props).root;
+
+    const memoizedShimmerData = useMemo(
+      () => ({
+        gradientTintColor: props.gradientTintColor ? props.gradientTintColor : tokens['gradientTintColor'],
+        shimmerTintColor: props.shimmerTintColor ? props.shimmerTintColor : tokens['shimmerTintColor'],
+        containerWidth: props.width ? props.width : tokens['width'],
+        containerHeight: props.height ? props.height : tokens['height'],
+        toValue: props.toValue ? props.toValue : tokens['toValue'],
+        delay: props.delay ? props.delay : tokens['delay'],
+        duration: props.duration ? props.duration : tokens['duration'],
+        angle: props.angle ? props.angle : tokens['angle'],
+        gradientOpacity: tokens['gradientOpacity'],
+      }),
+      [props.gradientTintColor, props.shimmerTintColor, props.width, props.height, props.toValue, props.delay, props.duration, props.angle, props.gradientOpacity],
+    );
+
+    let startValue = useShimmerAnimation(memoizedShimmerData);
+
+    return (rest: ShimmerProps) => {
+      const { circle, rect, uri, elements, ...mergedProps } = mergeProps(props, rest);
+      if (elements) {
+        var rows = [];
+        for (var i = 0; i < elements.length; i++) {
+          const element = elements[i];
+          if (element.type == ShimmerElementType.rect) {
+            rows.push(<Slots.rect key={i} width={element.width} height={element.height} x={element.xPos} y={element.yPos} rx={element.borderRadius} ry={element.borderRadius} />);
+          } else if (element.type == ShimmerElementType.circle) {
+            rows.push(<Slots.circle key={i} r={element.height / 2} cx={element.xPos} cy={element.yPos} />);
+          }
+        }
+      }
+      return (
+        <Slots.root {...mergedProps}>
+          <Defs>
```

Comment:
Very good point. Since startValue uses the state hook which causes re-render on every state change, this does cause performance issues. I learned that useRef does the same thing as useState - preserves the data after a re-render, except it doesn't cause re-render. Since we are not updating the component on state change, useRef is probably the more suitable approach here.

Reply:
Talked offline. I'm still not sure if rendering the animation redraws the SVG, opened https://github.com/microsoft/fluentui-react-native/issues/675  to track this.